### PR TITLE
Make foreign key property required and other relational properties optional

### PR DESCRIPTION
**Pull request by nabdelgadir**
Make foreign key property required and other relational properties optional

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: examples/todo-list/src/__tests__/acceptance/todo-list-todo.acceptance.ts [hunk 1]
```diff
@@ -42,7 +42,8 @@ describe('TodoListApplication', () => {
   });
 
   it('creates todo for a todoList', async () => {
-    const todo = givenTodo({todoListId: undefined});
+    const todo = givenTodo();
+    delete todo.todoListId;
```
Comment (bajtos) [inline examples/todo-list/src/__tests__/acceptance/todo-list-todo.acceptance.ts]: Why is it necessary to delete `todoListId` property, considering that it has been already specified as `todoListId: undefined` when calling `givenTodo`? Can we fix `givenTodo` to honor properties with `undefined` values instead?

File: docs/site/tutorials/todo-list/todo-list-tutorial-model.md [hunk 2]
```diff
@@ -85,7 +85,7 @@ export class TodoList extends Entity {
   // ...properties defined by the CLI...
 
   @hasMany(() => Todo)
-  todos?: Todo[];
+  todos: Todo[];
```
Comment (bajtos) [inline docs/site/tutorials/todo-list/todo-list-tutorial-model.md]: I think this property is supposed to be optional, because `todos` is often not defined (e.g. when returned by `find` call and no inclusion was configured). I am proposing to revert this change.

File: examples/todo-list/src/models/todo-list-image.model.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ export class TodoListImage extends Entity {
   id: number;
 
   @belongsTo(() => TodoList)
-  todoListId?: number;
+  todoListId: number;
```
Comment (bajtos) [inline examples/todo-list/src/models/todo-list-image.model.ts]: +1, the foreign key is usually required (`NOT NULL` in SQL).

File: docs/site/tutorials/todo-list/todo-list-tutorial-model.md (hunk 2 above)
Comment (nabdelgadir) [reply docs/site/tutorials/todo-list/todo-list-tutorial-model.md]: Ohh, I see, but this fix is just the docs was just to match the [code](https://github.com/strongloop/loopback-next/blob/verify-no-fk-required/examples/todo-list/src/models/todo-list.model.ts#L33). Should I change the actual code then?

File: examples/todo-list/src/models/todo-list-image.model.ts (hunk 3 above)
Comment (nabdelgadir) [reply examples/todo-list/src/models/todo-list-image.model.ts]: I'll revert this change.

File: examples/todo-list/src/__tests__/acceptance/todo-list-todo.acceptance.ts (hunk 1 above)
Comment (nabdelgadir) [reply examples/todo-list/src/__tests__/acceptance/todo-list-todo.acceptance.ts]: It does accept it, but this change was just to verify the property wasn't needed at all

File: examples/todo-list/src/models/todo-list-image.model.ts (hunk 3 above)
Comment (jannyHou) [reply examples/todo-list/src/models/todo-list-image.model.ts]: I might misunderstand something...
> the foreign key is usually required (NOT NULL in SQL).

Doesn't it mean the change is good 🙈

File: docs/site/tutorials/todo-list/todo-list-tutorial-model.md (hunk 2 above)
Comment (jannyHou) [reply docs/site/tutorials/todo-list/todo-list-tutorial-model.md]: > I think this property is supposed to be optional, because todos is often not defined (e.g. when returned by find call and no inclusion was configured). I am proposing to revert this change.

+1

Comment (bajtos) [reply docs/site/tutorials/todo-list/todo-list-tutorial-model.md]: Yes please, let's fix the code.

File: examples/todo-list/src/models/todo-list-image.model.ts (hunk 3 above)
Comment (bajtos) [reply examples/todo-list/src/models/todo-list-image.model.ts]: Yes, in my comment, I meant that I agreed with this proposed change too.

Please don't revert this change!

Comment (bajtos) [reply examples/todo-list/src/models/todo-list-image.model.ts]: Sorry for the confusion I created 🙈
